### DISCUSSION TITLE
How do I set x-model value on document load? Whatever I do I get document.getElementById(...)._x_model is undefined

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#3317] [1. Help]
How do I wait until Alpine is ready?

**Comment by ekwoka** [comment]
Alpine triggers multiple events. I think `apline:initialized` fires after init

**Comment by ghost** [comment]
@ekwoka could you please clarify? Let me describe my issue more precisely. 

Instead of using x-data I would like to use dynamically generated params based on GET params that the page gets.
I would like to set specific values as predefined for inputs with x-model.
As soon as I do in the browser's  console:
````
document.getElementById('edit-guests')._x_model.set(5);
````
Everything works fine. But if I add this code at the end of my document I get `document.getElementById(...)._x_model is undefined`
How can I overcome this?

**Comment by SimoTod** [comment]
You can also use x-init to make it a bit less disconnected (nextTick is important because x-init runs before x-model)
```html
<div x-data="{foo: 0}"?>
  <input x-model="foo" x-init="$nextTick(() => $el._x_model.set(2))">
</div>
```
